collector/collector.go

```diff
@@ -70,6 +70,7 @@ var (
 	// 64-bit float mantissa: https://en.wikipedia.org/wiki/Double-precision_floating-point_format
 	float64Mantissa uint64 = 9007199254740992
 	wrapCounters           = kingpin.Flag("snmp.wrap-large-counters", "Wrap 64-bit counters to avoid floating point rounding.").Default("true").Bool()
+	srcAddress             = kingpin.Flag("snmp.src-address", "Source ipv4 address to send snmp pdu.").Default("0").String()
```

Comment:
This should be implemented as a simple pass-through so all features of gosnmp `LocalAddress` can be used.

Plus it would be helpful to include the upstream doc string.

```suggestion
	srcAddress             = kingpin.Flag("snmp.source-address", "Source address to send snmp from in the format 'address:port' to use when connecting targets. If the port parameter is empty or '0', as in '127.0.0.1:' or '[::1]:0"', a source port number is automatically (random) chosen.").Default("").String()
```